### DISCUSSION TITLE
ELI5 Existing

### DISCUSSION THREAD

**Discussion by jrobber** [rnmapbox/maps#2905] [Q&A]
I don't understand what existing is referring to and why.  The deprecated docs page, assumes the developer is familiar with the problem and it's facets.  https://github.com/rnmapbox/maps/wiki/Deprecated-ExistingSourceLayer

If I add the existing prop to my ShapeSource I see this
```
Mapbox warning Warning source with id:collection-shape-source marked as existing, but could not find in style, source identifiers: [["composite", "vector"], ["rctmgl-mapview-point-annotations", "geojson"]] 
```

If I remove it I see this:
```
Mapbox warning Warning source with id:collection-shape-source referred to existing source but `existing` attibute was missing. https://github.com/rnmapbox/maps/wiki/Deprecated-ExistingSourceLayer {"level": "warning", "message": "Warning source with id:collection-shape-source referred to existing source but `existing` attibute was missing. 
```

My ShapeSource looks like this:
```
       <MapboxGL.ShapeSource
            id="collection-shape-source"
            shape={
              // featureCollection
              {
                type: 'FeatureCollection',
                features: collectionItemsArray,
              }
            }>
            <CircleLayer
              id="collection-item-layer"
              existing
              style={{
                circlePitchAlignment: 'map',
                circleColor: palette.Secondary_Blue_Base,
                circleRadius: 20,
                circleOpacity: 1,
                circleStrokeWidth: 1,
                circleStrokeColor: 'white',
              }}
            />
          </MapboxGL.ShapeSource>
```

Seems pretty straightforward to me

What properties are evaluated to determine if a source is or should be marked as "existing"?
What setup should be done to make this work correctly?

I have not used mapbox on web very much, I've been pretty focused on getting it working on Mobile.
Maybe more links from the RN docs to the main mapbox docs should be used when the RN docs are assuming pre-existing knowledge.

I'm pretty happy with RNMapBox, but learning it and getting it working is a real pain because I have to compare 2-3 sources of docs and figure out what's out of date and what's applicable on both.  

I guarantee it's hurting adoption.

**Comment by mfazekas** [comment]
@jrobber You need only use the `existing` if the layer/source is already is defined in the style you're using. In your example it should not be necessary. If you see that warning please file a bug report with a minimal but complete repro
